homeassistant/components/bosch_alarm/config_flow.py

```diff
@@ -108,6 +114,13 @@ async def async_step_user(
             else:
                 self._data = user_input
                 self._data[CONF_MODEL] = model
+
+                if self.source == SOURCE_RECONFIGURE:
+                    if (
+                        self._get_reconfigure_entry().data[CONF_MODEL]
+                        != self._data[CONF_MODEL]
+                    ):
+                        raise AbortFlow("device_mismatch")
```

Comment:
```suggestion
                        return self.async_abort(reason="device_mismatch")
```
while `AbortFlow` works, `async_abort` is more standard here